core/src/main/java/org/apache/iceberg/encryption/StandardEncryptionManager.java

```diff
@@ -41,7 +42,10 @@ public class StandardEncryptionManager implements EncryptionManager {
    * @param kmsClient Client of KMS used to wrap/unwrap keys in envelope encryption
    */
   public StandardEncryptionManager(
-      String tableKeyId, int dataKeyLength, KeyManagementClient kmsClient) {
+      String tableKeyId,
+      int dataKeyLength,
+      KeyManagementClient kmsClient,
+      boolean nativeDataEncryption) {
```

Comment:
I don't think that this should be passed in. The encryption manager needs to support files that use both native encryption (Parquet) and files that use AES GCM streams (Avro). There is no way to set this correctly because the behavior depends on the file type.

Instead, this should _always_ call `decrypt` and the caller should know how to handle the resulting `InputFile`. If it is simply an `InputFile` then it should be used directly. If it is a `StandardDecryptedInputFile` then it should be used for Avro but trigger native decryption for Parquet.